src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordClone.cs

```diff
@@ -0,0 +1,139 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CSharp.Emit;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SynthesizedRecordClone : SynthesizedInstanceMethodSymbol
+    {
+        public override NamedTypeSymbol ContainingType { get; }
+
+        public SynthesizedRecordClone(NamedTypeSymbol containingType)
+        {
+            ContainingType = containingType;
+        }
+        public override string Name => WellKnownMemberNames.CloneMethodName;
+
+        public override MethodKind MethodKind => MethodKind.Ordinary;
+
+        public override int Arity => 0;
+
+        public override bool IsExtensionMethod => false;
+
+        public override bool HidesBaseMethodsByName => false;
+
+        public override bool IsVararg => false;
+
+        public override bool ReturnsVoid => false;
+
+        public override bool IsAsync => false;
+
+        public override RefKind RefKind => RefKind.None;
+
+        public override ImmutableArray<ParameterSymbol> Parameters => ImmutableArray<ParameterSymbol>.Empty;
+
+        public override TypeWithAnnotations ReturnTypeWithAnnotations => TypeWithAnnotations.Create(
+            isNullableEnabled: true,
```

Comment:
>isNullableEnabled: true, [](start = 12, length = 24)

Why specify nullability information here, but return `None` from `ReturnTypeFlowAnalysisAnnotations` on line 50 below?

Reply:
I thought `None` was the appropriate value for something which has no nullable attributes applied. What should it be instead?

Reply:
Oops, I got confused. You're correct, `ReturnTypeFlowAnalysisAnnotations` is for the attributes.

---
In reply to: [432633596](https://github.com/dotnet/roslyn/pull/44582#discussion_r432633596) [](ancestors = 432633596)